protocols/kad/src/handler.rs

```diff
@@ -123,6 +128,9 @@ pub struct KademliaHandlerConfig {
 
     /// Time after which we close an idle connection.
     pub idle_timeout: Duration,
+
+    // If Mode::Client, node will act in `client` mode in the Kademlia network.
+    pub client: Mode,
```

Comment:
The `Mode` logic you are introducing in this pull request replaces the `KademliaHandlerConfig::allow_listening` logic. The latter can thus be replaced in favor of the former :tada: